src/components/fx/hover.js

```diff
@@ -264,9 +274,11 @@ function _hover(gd, evt, subplot, noHoverEvent) {
         if(hasUserCalledHover) {
             if('xpx' in evt) xpx = evt.xpx;
             else xpx = xaArray[0]._length / 2;
+            if(!('offsetX' in evt)) evt.offsetX = xpx + xaArray[0]._offset;
```

Comment:
Why do we need to fill `offsetX` and `offsetY` now?

Reply:
We need to fill it to provide data for the `spikesnap: cursor` case, so that `createspikes` function knows the coordinates of the pointer.
For clarity, I renamed these parameters to `pointerX` and `pointerY`.